WordPress/gutenberg
# Global Styles: Load block CSS conditionally

## What?
This is an alternative to https://github.com/WordPress/gutenberg/pull/40513. We should only load the Global Styles CSS for blocks that are used on the page, rather than loading it on every page as we do presently.

## Why?
This means sending less data on each page load, so it should improve performance. This is is a pre-cursor to https://github.com/WordPress/gutenberg/pull/34180, which will add a lot more CSS to each page load unless we improve the performance.

## How?
The basic premise is that if we are loading separate block assets, we move the block specific Global Styles CSS from the global-styles inline CSS block to the inline CSS portion of the block's CSS. The loading of this inline CSS is handled by wp_common_block_scripts_and_styles which enqueues the inline CSS for each block, so this just builds on top of that existing logic.

The changes we need to make are:

1. Don't add block specific CSS to the global styles inline CSS
2. Add add the block specific CSS to the individual block styles inline CSS

This differs from the approach in https://github.com/WordPress/gutenberg/pull/40513 by introducing a new filter for `get_style_nodes` which allows us to filter blocks from the style nodes used to generate the global styles, keeping WP_Theme_Json ignorant of the details of how assets are loaded.

Testing Instructions
1. Switch to TT2
2. Go to a page without a quote block on
3. Check that this CSS isn't in either the global-styles-inline-css or wp-block-quote-css:
.wp-block-quote {
border-width: 1px;
}
4. Go to a page with a quote block on
Check that this CSS is not in the global-styles-inline-css but is in wp-block-quote-css:
.wp-block-quote {
border-width: 1px;
}

## Note
The CSS for the blocks is doubled up - this is because its being enqueued in both core and by Gutenberg. This happens in trunk.

@WordPress/block-themers
